packages/genesys-spark-components/src/components/stable/gux-datepicker/gux-datepicker.tsx

```diff
@@ -158,6 +158,12 @@ export class GuxDatepicker {
   @State()
   active: boolean = false;
 
+  /**
+   * Tracks the amount of key presses when focusing on the input.
+   */
+  @State()
+  pressedKeys: Array<string> = [];
```

Comment:
Is there a reason to track the event keys in this array rather than just increment a number?

Reply:
So basically the `pressedKeys` array is being used to track the amount of keypresses on the input. For example if you focus on the `day` field of the input and press `1` you expect `01` then focus out of the input and focus on it again and you press `1` its expected to be `01` again.

Another scenario if you go into the day input and press `1` you expect `01` and then press `1` again expect `11`.

Basically in the V3 version of the datepicker this is not the case so we are trying to get it as close to the native date picker as possible.

If you have any other questions let me know.